### PR TITLE
android: use libc getifaddrs()

### PR DESCRIPTION
**Pull request by bnoordhuis**
Fixes: https://github.com/libuv/libuv/issues/3411

### CODE REVIEW

**Comment by bnoordhuis** [comment]
@gengjiawen I think I need some help tweaking CI.yml. I'm unclear on how to pick the right NDK version with reactnativecommunity/react-native-android.

**Comment by gengjiawen** [comment]
> @gengjiawen I think I need some help tweaking CI.yml. I'm unclear on how to pick the right NDK version with reactnativecommunity/react-native-android.

CI is using ndk `20.0.5594570`, which ndk version do you need ?

**Comment by gengjiawen** [comment]
Looks you forget bump minimum to 24 

```diff
--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -44,7 +44,7 @@ jobs:
         # see build options you can use in https://developer.android.com/ndk/guides/cmake
         run: |
           mkdir build && cd build
-          $ANDROID_HOME/cmake/3.10.2.4988404/bin/cmake -DCMAKE_TOOLCHAIN_FILE=$ANDROID_HOME/ndk/20.0.5594570/build/cmake/android.toolchain.cmake -DCMAKE_BUILD_TYPE=Release -DANDROID_ABI="arm64-v8a" -DANDROID_PLATFORM=android-21 ..
+          $ANDROID_HOME/cmake/3.10.2.4988404/bin/cmake -DCMAKE_TOOLCHAIN_FILE=$ANDROID_HOME/ndk/20.0.5594570/build/cmake/android.toolchain.cmake -DCMAKE_BUILD_TYPE=Release -DANDROID_ABI="arm64-v8a" -DANDROID_PLATFORM=android-24 ..
           $ANDROID_HOME/cmake/3.10.2.4988404/bin/cmake --build .
```

File: SUPPORTED_PLATFORMS.md [hunk 1]
```diff
@@ -11,7 +11,7 @@
 | z/OS | Tier 2 | >= V2R2 | Maintainers: @libuv/zos |
 | Linux with musl | Tier 2 | musl >= 1.0 | |
 | SmartOS | Tier 3 | >= 14.4 | |
-| Android | Tier 3 | NDK >= r15b | |
```
Comment (gengjiawen) [inline SUPPORTED_PLATFORMS.md:14]: r15b should still do. NDK is aligned with android sdk version. It has it's own version.

Comment (gengjiawen) [reply SUPPORTED_PLATFORMS.md:14]: SDK version: https://developer.android.com/studio/releases/platforms#7.0
NDK version: https://developer.android.com/ndk/downloads/revision_history

**Comment by bnoordhuis** [comment]
> Looks you forget bump minimum to 24

I was wondering if I also had to update the file path. Thanks for helping!

**Review by gengjiawen — APPROVED** [review]

File: SUPPORTED_PLATFORMS.md (hunk 1 above)
Comment (gengjiawen) [reply SUPPORTED_PLATFORMS.md]: Maybe add here we are now only supporting android 7.0 ?

Comment (bnoordhuis) [reply SUPPORTED_PLATFORMS.md:14]: Good idea, done.

**Review by cjihrig — APPROVED** [review]

**Pull request merged**